awx_collection/plugins/lookup/schedule_rruleset.py

```diff
@@ -0,0 +1,353 @@
+              - If this is a date in the format YYYY-MM-DD [HH:MM:SS], this schedule ends after this date
+              - Used for all types except none
+            type: str
+          bysetpos:
+            description:
+              - Specify an occurrence number, corresponding to the nth occurrence of the rule inside the frequency period.
+              - A comma-separated list of positions (first, second, third, forth or last)
+            type: string
+          bymonth:
+            description:
+              - The months this schedule will run on
+              - A comma-separated list which can contain values 0-12
+            type: string
+          bymonthday:
+            description:
+              - The day of the month this schedule will run on
+              - A comma-separated list which can contain values 0-31
+            type: string
+          byyearday:
+            description:
+              - The year day numbers to run this schedule on
+              - A comma-separated list which can contain values 0-366
+            type: string
+          byweekno:
+            description:
+              - The week numbers to run this schedule on
+              - A comma-separated list which can contain values as described in ISO8601
+            type: string
+          byweekday:
+            description:
+              - The days to run this schedule on
+              - A comma-separated list which can contain values sunday, monday, tuesday, wednesday, thursday, friday
+            type: string
+          byhour:
+            description:
+              - The hours to run this schedule on
+              - A comma-separated list which can contain values 0-23
+            type: string
+          byminute:
+            description:
+              - The minutes to run this schedule on
+              - A comma-separated list which can contain values 0-59
+            type: string
+          include:
+            description:
+              - If this rule should be included (RRULE) or excluded (EXRULE)
+            type: bool
+            default: True
+"""
+
+EXAMPLES = """
+    - name: Create a ruleset for everyday except Sundays
+      set_fact:
+        complex_rule: "{{ query(awx.awx.schedule_rruleset, '2022-04-30 10:30:45', rules=rrules, timezone='UTC' ) }}"
+      vars:
+        rrules:
+          - frequency: 'day'
+            interval: 1
+          - frequency: 'day'
+            interval: 1
+            byweekday: 'sunday'
+            include: False
+"""
+
+RETURN = """
+_raw:
+  description:
+    - String in the rrule format
+  type: string
+"""
+import re
+
+from ansible.module_utils.six import raise_from
+from ansible.plugins.lookup import LookupBase
+from ansible.errors import AnsibleError
+from datetime import datetime
+
+try:
+    import pytz
+    from dateutil import rrule
+except ImportError as imp_exc:
+    LIBRARY_IMPORT_ERROR = imp_exc
```

Comment:
Will this work right?  We're making use of `rrule` at the class scope below.

Reply:
Good catch, I moved the raise of the exception right into this except.